integrationtests/src/test/java/io/kroxylicious/proxy/KrpcFilterIT.java

```diff
@@ -0,0 +1,313 @@
+import org.apache.kafka.common.serialization.ByteArraySerializer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+
+import io.kroxylicious.proxy.config.ConfigParser;
+import io.kroxylicious.proxy.config.Configuration;
+import io.kroxylicious.proxy.internal.filter.ByteBufferTransformation;
+import io.kroxylicious.proxy.testkafkacluster.KafkaClusterConfig;
+import io.kroxylicious.proxy.testkafkacluster.KafkaClusterFactory;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class KrpcFilterIT {
+
+    private static final String TOPIC_1 = "my-test-topic";
+    private static final String TOPIC_2 = "other-test-topic";
+    private static final String PLAINTEXT = "Hello, world!";
+    private static final byte[] TOPIC_1_CIPHERTEXT = { (byte) 0x3d, (byte) 0x5a, (byte) 0x61, (byte) 0x61, (byte) 0x64, (byte) 0x21, (byte) 0x15, (byte) 0x6c,
+            (byte) 0x64, (byte) 0x67, (byte) 0x61, (byte) 0x59, (byte) 0x16 };
+    private static final byte[] TOPIC_2_CIPHERTEXT = { (byte) 0xffffffa7, (byte) 0xffffffc4, (byte) 0xffffffcb, (byte) 0xffffffcb, (byte) 0xffffffce, (byte) 0xffffff8b,
+            (byte) 0x7f, (byte) 0xffffffd6, (byte) 0xffffffce, (byte) 0xffffffd1, (byte) 0xffffffcb, (byte) 0xffffffc3, (byte) 0xffffff80 };
+
+    private TestInfo testInfo;
+
+    @BeforeEach
+    public void beforeEach(TestInfo testInfo) {
+        this.testInfo = testInfo;
+    }
+
+    @BeforeAll
+    public static void checkReversibleEncryption() {
+        // The precise details of the cipher don't matter
+        // What matters is that it the ciphertext key depends on the topic name
+        // and that decode() is the inverse of encode()
+        assertArrayEquals(TOPIC_1_CIPHERTEXT, encode(TOPIC_1, ByteBuffer.wrap(PLAINTEXT.getBytes(StandardCharsets.UTF_8))).array());
+        assertEquals(PLAINTEXT, new String(decode(TOPIC_1, ByteBuffer.wrap(TOPIC_1_CIPHERTEXT)).array(), StandardCharsets.UTF_8));
+        assertArrayEquals(TOPIC_2_CIPHERTEXT, encode(TOPIC_2, ByteBuffer.wrap(PLAINTEXT.getBytes(StandardCharsets.UTF_8))).array());
+        assertEquals(PLAINTEXT, new String(decode(TOPIC_2, ByteBuffer.wrap(TOPIC_2_CIPHERTEXT)).array(), StandardCharsets.UTF_8));
+    }
+
+    public static class TestEncoder implements ByteBufferTransformation {
+
+        @Override
+        public ByteBuffer transform(String topicName, ByteBuffer in) {
+            return encode(topicName, in);
+        }
+    }
+
+    private static ByteBuffer encode(String topicName, ByteBuffer in) {
+        var out = ByteBuffer.allocate(in.limit());
+        byte rot = (byte) (topicName.hashCode() % Byte.MAX_VALUE);
+        // System.out.println("Encode Rot " + rot);
```

Comment:
Should we remove this debug line? Same as below.

Reply:
Done